src/commands/auth/login.ts

```diff
@@ -1,43 +1,71 @@
-import {Command, ux} from '@oclif/core'
+import {Command, Flags, ux} from '@oclif/core'
 import * as Fs from 'node:fs'
 import * as toml from '@iarna/toml'
 import {getGrowthBookConfigDirectory, getGrowthBookConfigFilePath} from '../../utils/file'
 import {DEFAULT_GROWTHBOOK_BASE_URL, DEFAULT_GROWTHBOOK_PROFILE} from '../../utils/constants'
-import {Icons} from '../../utils/cli'
+import {Icons, baseGrowthBookCliFlags} from '../../utils/cli'
+import {getGrowthBookProfileConfigAndThrowForCommand} from '../../utils/config'
 
 export default class Login extends Command {
   static description = 'Configure your API key with the GrowthBook SDK with your project'
 
   static examples = []
 
-  static flags = {}
+  static flags = {
+    ...baseGrowthBookCliFlags,
+    apiKey: Flags.string({
+      description: 'Your GrowthBook secret API key',
+      required: false,
+    }),
+  }
 
   static args = {}
 
   async run(): Promise<void> {
-    const apiKey = await ux.prompt('What is your GrowthBook secret API Key?', {
-      type: 'hide',
-      required: true,
-    })
-    if (!apiKey) {
-      this.error(`${Icons.xSymbol} You must provide a GrowthBook secret API key to continue`)
+    const {
+      args: {
+        apiKey,
+      },
+      flags: {
+        profile,
+        apiBaseUrl,
+      },
+    } = await this.parse(Login)
+    let profileUsed = profile || DEFAULT_GROWTHBOOK_PROFILE
```

Comment:
This looks like if they don't pass in a profile as a flag, we will end up overwriting their default profile, and with this logic we will never prompt them for a profile below.

Reply:
Good catch, fixed!